rmgpy/pdep/cse.pyx

```diff
@@ -217,9 +217,168 @@ def apply_chemically_significant_eigenvalues_method(network, list lumping_order=
                     index = indices[i, r, s]
                     if index > -1:
                         pa[i, src1, r, s] = x[index, src]
-    
+
     pa[:, n_isom:, :, :] /= ym_b
     k[:, n_isom:] /= ym_b
 
     # Return the matrix of k(T,P) values and the pseudo-steady population distributions
     return k, pa
+
+def get_rate_coefficients_CSE_Advanced(network, T, P, neglect_high_energy_collisions=False, high_energy_rate_tol=0.01):
+    """
+    CSE using the Georgievskii et al. 2013 method https://doi.org/10.1021/jp4060704
+    """
+    if network.T != T or network.P != P:
+        network.set_conditions(T,P)
+        network.calculate_equilibrium_ratios()
+
+    lumping_order = None
+
+    temperature = network.T
+    pressure = network.P
+    beta = 1.0/(constants.R*temperature)
+    e_list = network.e_list
+    j_list = network.j_list
+    dens_states = network.dens_states
+    g_nj = network.Gnj
+    f_im = network.Fim
+    eq_ratios = network.eq_ratios
+    n_isom = network.n_isom
+    n_reac = network.n_reac
+    n_prod = network.n_prod
+    n_grains = network.n_grains
+    n_j = network.n_j
+    n_channels = n_isom+n_reac
+
+    channels = network.isomers+network.reactants
+
+    n_grains = len(e_list)
+    n_chem = n_isom #+ n_reac
+
+
+    # Generate the full master equation matrix
+    me_mat, indices = generate_full_me_matrix(network, products=False, exclude_association=True,
+                                                  neglect_high_energy_collisions=neglect_high_energy_collisions,
+                                                  high_energy_rate_tol=high_energy_rate_tol)
+    n_rows = me_mat.shape[0]
+
+    # Generate symmetrization matrix and its inverse
+    s_mat = np.zeros(n_rows, np.float64)
+    s_mat_inv = np.zeros_like(s_mat)
+    for i in range(n_isom):
+        for r in range(n_grains):
+            for s in range(n_j):
+                index = indices[i, r, s]
+                if index > -1:
+                    s_mat[index] = sqrt(dens_states[i, r, s] * (2 * j_list[s] + 1) \
+                                            * exp(-e_list[r] / (constants.R * temperature)) * eq_ratios[i])
+                    s_mat_inv[index] = 1.0 / s_mat[index]
+
+    # Symmetrize master equation matrix: me_mat = s_mat * m_symm * s_mat_inv
+    # Since s_mat and s_mat_inv are diagonal we can do this very efficiently
+    for r in range(n_rows):
+        for s in range(n_rows):
+            me_mat[r, s] = s_mat_inv[r] * me_mat[r, s] * s_mat[s]
+
+    # DEBUG: Check that the matrix has been properly symmetrized
```

Comment:
Why DEBUG here?

Reply:
This code segment is copied from the vanilla cse method. Fundamentally, it's running a code check that should be true there so I guess they called it a debugging opeeration.